### PR TITLE
rds_cluster: add support for removing cluster from global db

### PR DESCRIPTION
**Pull request by mandar242**
##### SUMMARY

add support for removing cluster from global db

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

rds_cluster
##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: plugins/modules/rds_cluster.py [hunk 1]
```diff
@@ -463,6 +472,13 @@
     engine: aurora-postgresql
     state: present
     db_instance_class: 'db.t3.medium'
+
+- name: Remove a cluster from global DB
```
Comment (alinabuzachis) [inline plugins/modules/rds_cluster.py:476]: What if the user wants to delete the cluster and remove It from the global db cluster at the same time?

Comment (mandar242) [reply plugins/modules/rds_cluster.py:476]: if deleting a cluster which is part of a global db, then it fails with boto error (cannot delete, part of global db.....)

so with current changes in PR, users can use 1 task to remove cluster from global db and another task to delete the cluster

Comment (alinabuzachis) [reply plugins/modules/rds_cluster.py:475]: How much does it take to remove the rds cluster from a global one?

File: tests/integration/targets/rds_cluster_modify/tasks/main.yaml [hunk 2]
```diff
@@ -4,7 +4,16 @@
       aws_access_key: '{{ aws_access_key }}'
       aws_secret_key: '{{ aws_secret_key }}'
       security_token: '{{ security_token | default(omit) }}'
+    group/amazon.cloud.aws:
```
Comment (GomathiselviS) [inline tests/integration/targets/rds_cluster_modify/tasks/main.yaml:7]: Why is this needed here?

File: plugins/modules/rds_cluster.py (hunk 1 above)
Comment (mandar242) [reply plugins/modules/rds_cluster.py:475]: from my trial-errors, it takes a couple of minutes

File: tests/integration/targets/rds_cluster_modify/tasks/main.yaml (hunk 2 above)
Comment (mandar242) [reply tests/integration/targets/rds_cluster_modify/tasks/main.yaml:7]: for the tasks in integration tests that use `amazon.cloud` collection
like https://github.com/ansible-collections/amazon.aws/pull/1705/files#diff-91943e165e49a6da17ee2d4839a338f90e72c809e45c6772437bf3f62d383664R5-R12

Comment (GomathiselviS) [reply tests/integration/targets/rds_cluster_modify/tasks/main.yaml:7]: I think you need to use the collections keyword to list the amazon.cloud collection.

Comment (mandar242) [reply tests/integration/targets/rds_cluster_modify/tasks/main.yaml:7]: I don't think that is needed, from my testing it looks like, if I remove that part, the the tasks that use `amazon.cloud` fail with `credentials not found`

```
TASK [rds_cluster_modify : Create rds global database] *************************
An exception occurred during task execution. To see the full traceback, use -vvv. The error was: botocore.exceptions.NoCredentialsError: Unable to locate credentials
fatal: [testhost]: FAILED! => {"boto3_version": "1.22.0", "botocore_version": "1.27.82", "changed": false, "msg": "Failed to modify resource: Unable to locate credentials", "resource_actions": ["cloudcontrolapi:GetResource"]}
```

File: plugins/modules/rds_cluster.py [hunk 3]
```diff
@@ -103,6 +103,9 @@
         aliases:
           - db_name
         type: str
+    db_cluster_arn:
```
[the remark this answers is not in the document]
Comment (mandar242) [reply plugins/modules/rds_cluster.py:106]: fixed https://github.com/ansible-collections/amazon.aws/pull/1705/files#diff-41d80755edc72340b8c75ea1d8c215e8e9b974de3ffe70e2bd0538c37baa2f0aR1132-R1135

File: plugins/modules/rds_cluster.py [hunk 4]
```diff
@@ -1102,6 +1117,28 @@ def ensure_present(cluster, parameters, method_name, method_options_name):
...
+
+    try:
+        client.remove_from_global_cluster(DbClusterIdentifier=db_cluster_arn, GlobalClusterIdentifier=global_cluster_id)
+        changed = True
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        module.fail_json_aws(
+            e, msg=f"Failed to remove cluster {db_cluster_id} from global DB cluster {global_cluster_id}."
+        )
+
+    module.exit_json(
```
[the remark this answers is not in the document]
Comment (mandar242) [reply plugins/modules/rds_cluster.py]: Updated the code logic to `not exit` after removing cluster from global db.
now the code continues to perform any needed operation such as modify or delete cluster.

File: tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml [hunk 5]
```diff
@@ -0,0 +1,246 @@
...
+          - modify_port_result is changed
+          - replica_cluster_info_result.port == 3389
+
+    - name: Get global db info
+      command: 'aws rds describe-global-clusters --global-cluster-identifier {{ test_global_cluster_name }}'
+      environment:
+        AWS_ACCESS_KEY_ID: "{{ aws_access_key }}"
+        AWS_SECRET_ACCESS_KEY: "{{ aws_secret_key }}"
+        AWS_SESSION_TOKEN: "{{ security_token | default('') }}"
+        AWS_DEFAULT_REGION: "{{ test_primary_cluster_region }}"
```
Comment (alinabuzachis) [inline tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml:145]: Same.

File: tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml [hunk 6]
```diff
@@ -0,0 +1,246 @@
...
+        remove_from_global_db: true
+
+    - name: Get global db info
+      command: 'aws rds describe-global-clusters --global-cluster-identifier {{ test_global_cluster_name }}'
+      environment:
+        AWS_ACCESS_KEY_ID: "{{ aws_access_key }}"
+        AWS_SECRET_ACCESS_KEY: "{{ aws_secret_key }}"
+        AWS_SESSION_TOKEN: "{{ security_token | default('') }}"
+        AWS_DEFAULT_REGION: "{{ test_primary_cluster_region }}"
+      register: global_cluster_info_result
```
Comment (alinabuzachis) [inline tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml:192]: Same.

File: plugins/modules/rds_cluster.py [hunk 7]
```diff
@@ -1102,6 +1129,29 @@ def ensure_present(cluster, parameters, method_name, method_options_name):
...
+        client.remove_from_global_cluster(DbClusterIdentifier=db_cluster_arn, GlobalClusterIdentifier=global_cluster_id)
+    except (botocore.exceptions.BotoCoreError, botocore.exceptions.ClientError) as e:
+        module.fail_json_aws(
+            e, msg=f"Failed to remove cluster {db_cluster_id} from global DB cluster {global_cluster_id}."
+        )
+
+    # this is needed as even though we have wait_for_cluster_status, it takes a few seconds
+    # for cluster to change status from 'available' to 'promoting'
+    time.sleep(15)
+    wait_for_cluster_status(client, module, db_cluster_id, "cluster_available")
```
Comment (alinabuzachis) [inline plugins/modules/rds_cluster.py:1150]: With this change, the user has to wait no matter what even if there is no change to apply to the cluster or maybe they do not want to wait. Maybe I am wrong, but it would perhaps be more efficient, if there is no change to apply to the cluster, but only to remove from the global db and the user does not want to wait, just remove it without any waiting. On the other hand, if there is something to be updated or if there is nothing to be updated and the user wants to wait for the cluster to be available, in that case you have to wait. I might be wrong, but since rds_cluster is s slow module, I guess we should probably minimise the waiting times when possible, this is my thought.

File: tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml [hunk 8]
```diff
@@ -0,0 +1,246 @@
...
+      register: primary_cluster_info_result
+
+    - name: Get global db info
+      command: 'aws rds describe-global-clusters --global-cluster-identifier {{ test_global_cluster_name }}'
+      environment:
+        AWS_ACCESS_KEY_ID: "{{ aws_access_key }}"
+        AWS_SECRET_ACCESS_KEY: "{{ aws_secret_key }}"
+        AWS_SESSION_TOKEN: "{{ security_token | default('') }}"
+        AWS_DEFAULT_REGION: "{{ test_primary_cluster_region }}"
+      register: global_cluster_info_result
```
[the remark this answers is not in the document]
Comment (mandar242) [reply tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml:49]: I had tried using `amazon.cloud.rds_global_cluster`, but none of the states `list`, `get`, `describe` return the info on Global Cluster's Members

tried these
```
    - name: get global cluster info
      amazon.cloud.rds_global_cluster:
        global_cluster_identifier: ansible-test-global-xxxx
        region: us-west-2
        state: get
      register: result1
    - debug: var=result1

    - name: describe global cluster info
      amazon.cloud.rds_global_cluster:
        global_cluster_identifier: ansible-test-global-xxxx
        region: us-west-2
        state: describe
      register: result2
    - debug: var=result2

    - name: list global cluster info
      amazon.cloud.rds_global_cluster:
        global_cluster_identifier: ansible-test-global-xxxx
        region: us-west-2
        state: list
      register: result3
    - debug: var=result3
```

all three return the same, doesn't have info on member clusters, shows arn of primary cluster as `source_db_cluster_identifier` but nothing about replica cluster
```
{
            "identifier": "ansible-test-global-xxxx",
            "properties": {
                "deletion_protection": false,
                "engine": "aurora-mysql",
                "engine_version": "5.7.mysql_aurora.2.11.2",
                "global_cluster_identifier": "ansible-test-global-xxxx",
                "source_db_cluster_identifier": "arn:aws:rds:us-west-2:xxxxxxxxxxxx:cluster:ansible-test-primary-xxxx",
                "storage_encrypted": false
}
```

File: tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml [hunk 9]
```diff
@@ -0,0 +1,246 @@
...
+      register: replica_cluster_info_result
+
+    - name: Get global db info
+      command: 'aws rds describe-global-clusters --global-cluster-identifier {{ test_global_cluster_name }}'
+      environment:
+        AWS_ACCESS_KEY_ID: "{{ aws_access_key }}"
+        AWS_SECRET_ACCESS_KEY: "{{ aws_secret_key }}"
+        AWS_SESSION_TOKEN: "{{ security_token | default('') }}"
+        AWS_DEFAULT_REGION: "{{ test_primary_cluster_region }}"
+      register: global_cluster_info_result
```
[the remark this answers is not in the document]
Comment (mandar242) [reply tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml:85]: https://github.com/ansible-collections/amazon.aws/pull/1705#discussion_r1316277219

File: tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml (hunk 5 above)
Comment (mandar242) [reply tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml:145]: https://github.com/ansible-collections/amazon.aws/pull/1705#discussion_r1316277219

File: tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml (hunk 6 above)
Comment (mandar242) [reply tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml:192]: https://github.com/ansible-collections/amazon.aws/pull/1705#discussion_r1316277219

File: plugins/modules/rds_cluster.py (hunk 7 above)
Comment (mandar242) [reply plugins/modules/rds_cluster.py:1150]: imo, 

1. the part `time.sleep(15)`, is much needed as without the sleep time, integration tests return false failure due to time it takes for clusterto go `from available to promoting` state.

2. the `wait_for_cluster_status(client, module, db_cluster_id, "cluster_available")` is also something I would like to have, as if there is any changes to apply to cluster along with removing it from global db, the changes won't get applied while the cluster is in `promoting` state.

I could be wrong though. cc @hakbailey @jillr

File: plugins/modules/rds_cluster.py [hunk 10]
```diff
@@ -305,6 +308,12 @@
         aliases:
           - maintenance_window
         type: str
+    remove_from_global_db:
+        description:
+          - If set to true, the cluster will be removed from global DB.
+          - Parameters global_cluster_identifier, db_cluster_identifier must be specified when I(remove_from_global_db=true).
+        type: bool
+        required: False
```
[the remark this answers is not in the document]
Comment (mandar242) [reply plugins/modules/rds_cluster.py:316]: added

File: plugins/modules/rds_cluster.py (hunk 7 above)
Comment (alinabuzachis) [reply plugins/modules/rds_cluster.py:1151]: What I'm suggesting is to wait only when needed.

Comment (mandar242) [reply plugins/modules/rds_cluster.py:1151]: okay, I'll change it to wait when `wait` param is set `true`
https://github.com/ansible-collections/amazon.aws/blob/main/plugins/modules/rds_cluster.py#L60

Comment (alinabuzachis) [reply plugins/modules/rds_cluster.py]: @jillr @tremble  any thoughts?

Comment (jillr) [reply plugins/modules/rds_cluster.py]: We chatted about this and I think we can add a custom waiter to https://github.com/ansible-collections/amazon.aws/blob/main/plugins/module_utils/waiters.py that would cover this scenario without having a have a static sleep.  @mandar242 will give that a try,

Comment (mandar242) [reply plugins/modules/rds_cluster.py]: @jillr @alinabuzachis
Added a custom waiter that - 'waits for promoting initiates on AWS end, i.e. cluster status changes from `available` to `promoting`'.
https://github.com/ansible-collections/amazon.aws/pull/1705/commits/eedaa256c9c39bfb745f4f06b95261772e6c12d2

**Comment by mandar242** [comment]
recheck

File: plugins/modules/rds_cluster.py [hunk 11]
```diff
@@ -305,6 +305,15 @@
         aliases:
           - maintenance_window
         type: str
+    remove_from_global_db:
+        description:
+          - If set to C(true), the cluster will be removed from global DB.
+          - Parameters I(global_cluster_identifier), I(db_cluster_identifier) must be specified when I(remove_from_global_db=true).
+          - When performing other modifications along with I(remove_grom_global_db=true), must set I(wait=true) as DB cluster
```
Comment (jillr) [inline plugins/modules/rds_cluster.py:312]:
```suggestion
          - When performing other modifications along with I(remove_from_global_db=true), must set I(wait=true) as DB cluster
```

File: plugins/modules/rds_cluster.py [hunk 12]
```diff
@@ -305,6 +308,11 @@
         aliases:
           - maintenance_window
         type: str
+    remove_from_global_db:
+        description:
```
[the remark this answers is not in the document]
Comment (jillr) [reply plugins/modules/rds_cluster.py:309]: @mandar242 Can you please add a required_if for the new option needing global_cluster_identifier and db_cluster_identifier?  This lets us catch the missing option faster for the user, and give them the clearest possible error message about the missing required parameter if they don't include the IDs.

The wait parameter being required when also making a change is probably too complex a scenario to cover with a required_if check so the docs should suffice for that.

File: plugins/modules/rds_cluster.py (hunk 11 above)
Comment (mandar242) [reply plugins/modules/rds_cluster.py]: Fixed.

File: tests/integration/targets/rds_cluster_modify/tasks/main.yaml [hunk 13]
```diff
@@ -177,7 +183,7 @@
 
   - name: Create DB cluster parameter group if not exists
     command: aws rds create-db-cluster-parameter-group --db-cluster-parameter-group-name
-      {{ new_db_parameter_group_name }} --db-parameter-group-family aurora-mysql5.7 --description
+      {{ new_db_parameter_group_name }} --db-parameter-group-family aurora-mysql8.0 --description
```
Comment (mandar242) [inline tests/integration/targets/rds_cluster_modify/tasks/main.yaml:186]: note: updated the engine version due to following failure seen in integration tests

```

2023-09-11 20:41:13.346386 | controller |   "msg": "Unable to modify DB cluster: An error occurred 
(InvalidParameterCombination) when calling the ModifyDBCluster operation: The Parameter Group 
ansible-test-db-parameter-group-e096d2ecba72-new with DBParameterGroupFamily aurora-mysql5.7 
cannot be used for this instance. Please use a Parameter Group with DBParameterGroupFamily
 aurora-mysql8.0",


```

File: tests/integration/targets/rds_cluster_modify/tasks/main.yaml [hunk 14]
```diff
@@ -5,6 +5,12 @@
       secret_key: '{{ aws_secret_key }}'
       session_token: '{{ security_token | default(omit) }}'
   block:
+
```
Comment (GomathiselviS) [inline tests/integration/targets/rds_cluster_modify/tasks/main.yaml:8]: Don't we need group/amazon.cloud.aws under module_defaults for the amazon.cloud module to work?

Comment (mandar242) [reply tests/integration/targets/rds_cluster_modify/tasks/main.yaml:8]: Had discussed this with Jill, it worked without `group/amazon.cloud.aws` after adding `amazon.cloud` to test requirements
https://github.com/ansible-collections/amazon.aws/pull/1705/files#diff-12ceb68f7a1282d864ebd30d4e5732b76b32fe7d3bdf77969043694f5b6346dfL3-R5

**Comment by GomathiselviS** [comment]
when I tried to run the integration tests locally, the following error was seen. 

```
"error": {
        "code": "InvalidVPCNetworkStateFault",
        "message": "The DB subnet group doesn't meet Availability Zone (AZ) coverage requirement. Current AZ coverage: us-east-1f. Add subnets to cover at least 2 AZs.",
        "type": "Sender"
    },
 ```

**Comment by mandar242** [comment]
> when I tried to run the integration tests locally, the following error was seen.
> 
> ```
> "error": {
>         "code": "InvalidVPCNetworkStateFault",
>         "message": "The DB subnet group doesn't meet Availability Zone (AZ) coverage requirement. Current AZ coverage: us-east-1f. Add subnets to cover at least 2 AZs.",
>         "type": "Sender"
>     },
> ```

@GomathiselviS 
so there are a couple of things
1. the rds_cluster module when creating db clusters, uses default vpc / subnet in the specified region, so it would be best to use regions that have default VPCs
2. maybe your local setup for credentials is somehow overriding the regions set in the test here https://github.com/ansible-collections/amazon.aws/pull/1705/files#diff-895307bb23ed08428be51129ac5c2cdc2eea1489bb300ea5c462cfa8fd7e162bR25-R30
i.e. `us-west-2`, `eu-north-1`. I had to remove `region` from my `~/.aws/credentials` to make sure it doesn't override regions specified in test.

**Comment by GomathiselviS** [comment]
> > when I tried to run the integration tests locally, the following error was seen.
> > ```
> > "error": {
> >         "code": "InvalidVPCNetworkStateFault",
> >         "message": "The DB subnet group doesn't meet Availability Zone (AZ) coverage requirement. Current AZ coverage: us-east-1f. Add subnets to cover at least 2 AZs.",
> >         "type": "Sender"
> >     },
> > ```
> 
> @GomathiselviS so there are a couple of things
> 
> 1. the rds_cluster module when creating db clusters, uses default vpc / subnet in the specified region, so it would be best to use regions that have default VPCs
> 2. maybe your local setup for credentials is somehow overriding the regions set in the test here https://github.com/ansible-collections/amazon.aws/pull/1705/files#diff-895307bb23ed08428be51129ac5c2cdc2eea1489bb300ea5c462cfa8fd7e162bR25-R30
>    i.e. `us-west-2`, `eu-north-1`. I had to remove `region` from my `~/.aws/credentials` to make sure it doesn't override regions specified in test.

I believe it would be beneficial to explicitly provide the 'region' parameter in the tasks and assign it to the variable used in default/main.yaml, in order to ensure that the tests use the region mentioned in that file. Currently, when running the tests using 'cloud-config-aws.ini', the aws_region is picked up from the ini file. However, in some of the tasks in the tests, the region is explicitly specified. like [here](https://github.com/ansible-collections/amazon.aws/pull/1705/files#diff-91943e165e49a6da17ee2d4839a338f90e72c809e45c6772437bf3f62d383664R48)

File: tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml [hunk 15]
```diff
@@ -0,0 +1,244 @@
...
+        db_instance_identifier: "{{ test_primary_cluster_name }}-instance"
+        region: "{{ test_primary_cluster_region }}"
+        engine: "{{ test_engine }}"
+        db_instance_class: "{{ test_instance_class }}"
+
+    - name: Get primary cluster info
+      amazon.aws.rds_cluster:
+        global_cluster_identifier: "{{ test_global_cluster_name }}"
+        db_cluster_identifier: "{{ test_primary_cluster_name }}"
+      register: primary_cluster_info_result
```
Comment (GomathiselviS) [inline tests/integration/targets/rds_cluster_modify/tasks/remove_from_global_db.yaml:40]: Can amazon.aws.rds_cluster_info module be used for this task? Seems like with the above task, amazon.aws.rds_cluster calls create_db_cluster and it fails with Parameter validation failed:\nMissing required parameter in input: \"Engine\"",

**Review by GomathiselviS — APPROVED** [review]

**Comment by mandar242** [comment]
recheck

**Comment by mandar242** [comment]
recheck

File: plugins/modules/rds_cluster.py [hunk 16]
```diff
@@ -1245,12 +1317,17 @@ def main():
...
-    parameters = arg_spec_to_rds_params(dict((k, module.params[k]) for k in module.params if k in parameter_options))
     changed = False
+
+    parameters = arg_spec_to_rds_params(dict((k, module.params[k]) for k in module.params if k in parameter_options))
     method_name, method_options_name = get_rds_method_attribute_name(cluster)
 
     if method_name:
         if method_name == "delete_db_cluster":
+            if module.params.get("remove_from_global_db"):
+                if cluster["Engine"] in ["aurora", "aurora-mysql", "aurora-postgresql"]:
```
Comment (GomathiselviS) [inline plugins/modules/rds_cluster.py:1328]: This will thrown an error, when we try to delete a cluster that is not present.

Comment (mandar242) [reply plugins/modules/rds_cluster.py:1328]: from my testing, doesn't seem to throw error.
but added `if cluster` for extra check

**Review by alinabuzachis — APPROVED** [review]

**Comment by alinabuzachis** [comment]
recheck

**Comment by mandar242** [comment]
recheck

File: plugins/modules/rds_cluster.py [hunk 17]
```diff
@@ -463,6 +470,42 @@
...
+- name: Remove a cluster from global DB and Delete without creating a final snapshot
+  amazon.aws.rds_cluster:
+    engine: aurora
+    password: "{{ password }}"
+    username: "{{ username }}"
+    cluster_id: "{{ cluster_id }}"
+    skip_final_snapshot: true
+    remove_from_global_db: true
+    wait: true
+    state: absent
```
Comment (GomathiselviS) [inline plugins/modules/rds_cluster.py:489]: This is resulting in the following error - `botocore.exceptions.WaiterError: Waiter cluster_available failed: Max attempts exceeded. Previously accepted state: Matched expected service error code: DBClusterNotFoundFault`

The task is 
`- name: Delete replica cluster
      amazon.aws.rds_cluster:
        db_cluster_identifier: "{{ test_replica_cluster_name }}"
        global_cluster_identifier: "{{ test_global_cluster_name }}"
        engine: "{{ rds_engine }}"
        engine_version: "{{ rds_engine_version }}"
        skip_final_snapshot: true
        remove_from_global_db: true
        region: "{{ test_replica_cluster_region }}"
        state: absent`
        
      Shouldn't this task work?

Comment (mandar242) [reply plugins/modules/rds_cluster.py:489]: it should work, can you retry running the tests? I've seen aws being flakey at times on this
I'll try running it again as well from my end and keep you posted

Comment (GomathiselviS) [reply plugins/modules/rds_cluster.py:489]: I've noticed that the first attempt at running it always fails, but it seems to pass when run a second time. The issue is consistent, so it's better to address it and find a solution to prevent this from happening.

Comment (mandar242) [reply plugins/modules/rds_cluster.py:489]: i'm not sure what can be done to address aws side errors, maybe increasing the retries and wait time?
in waiter here https://github.com/ansible-collections/amazon.aws/pull/1705/files#diff-9eb5385261889b20f465ea56fd4fe899498b77edc95f084f8b14c2075052c3d7R550-R554

Comment (mandar242) [reply plugins/modules/rds_cluster.py:489]: @GomathiselviS increase wait time fro. 60 seconds to 300, should work now

Comment (mandar242) [reply plugins/modules/rds_cluster.py:489]: worked fine, tests passed in first run for me

```
TASK [rds_cluster_modify : Assert that primary cluster is NOT a part of global db] ***
ok: [testhost] => {
    "changed": false,
    "msg": "All assertions passed"
}

TASK [rds_cluster_modify : Delete replica cluster] *****************************
ok: [testhost] => {"changed": false, "resource_actions": ["rds:DescribeDBClusters"]}

TASK [rds_cluster_modify : Delete instance connected to primary cluster] *******
changed: [testhost] => {"changed": true, "resource_actions": ["rds:ListTagsForResource", "rds:DeleteDBInstance", "rds:DescribeDBInstances"]}

TASK [rds_cluster_modify : Delete primary cluster] *****************************
changed: [testhost] => {"changed": true, "resource_actions": ["rds:DeleteDBCluster", "rds:DescribeDBClusters"]}

TASK [rds_cluster_modify : Delete global db] ***********************************
changed: [testhost] => {"changed": true, "resource_actions": ["cloudcontrolapi:GetResource", "cloudcontrolapi:ListResourceRequests", "cloudcontrolapi:DeleteResource"], "result": {}}

PLAY RECAP *********************************************************************
testhost                   : ok=31   changed=14   unreachable=0    failed=0    skipped=0    rescued=0    ignored=1   

AWS ACTIONS: ['cloudcontrolapi:CreateResource', 'cloudcontrolapi:DeleteResource', 'cloudcontrolapi:GetResource', 'cloudcontrolapi:GetResourceRequestStatus', 'cloudcontrolapi:ListResourceRequests', 'rds:CreateDBCluster', 'rds:CreateDBInstance', 'rds:DeleteDBCluster', 'rds:DeleteDBInstance', 'rds:DescribeDBClusters', 'rds:DescribeDBInstances', 'rds:ListTagsForResource', 'rds:ModifyDBCluster', 'rds:RemoveFromGlobalCluster']

```

File: plugins/modules/rds_cluster.py [hunk 18]
```diff
@@ -1261,6 +1338,12 @@ def main():
     else:
         cluster_id = module.params["db_cluster_identifier"]
 
+    if cluster and module.params.get("remove_from_global_db"):
+        if cluster["Engine"] in ["aurora", "aurora-mysql", "aurora-postgresql"]:
+            if changed:
+                wait_for_cluster_status(client, module, cluster_id, "cluster_available")
```
Comment (GomathiselviS) [inline plugins/modules/rds_cluster.py:1344]: This line will throw `botocore.exceptions.WaiterError: Waiter cluster_available failed: Max attempts exceeded. Previously accepted state: Matched expected service error code: DBClusterNotFoundFault` when state is absent and remove_from_global_db is true.

Comment (mandar242) [reply plugins/modules/rds_cluster.py:1344]: I don't think that should happen, as the call to `remove from global db` takes place before it makes call to `delete` the cluster

Comment (mandar242) [reply plugins/modules/rds_cluster.py:1344]: I had added these as suggested by @alinabuzachis

Comment (mandar242) [reply plugins/modules/rds_cluster.py]: @GomathiselviS made the fix, should work now

**Review by jillr — APPROVED** [review]

**Comment by mandar242** [comment]
recheck
